logic/src/commonMain/kotlin/com/wire/kalium/logic/data/notification/LocalNotificationMessageMapper.kt

```diff
@@ -103,4 +111,21 @@ class LocalNotificationMessageMapperImpl : LocalNotificationMessageMapper {
             messageContent.editMessageId,
             LocalNotificationUpdateMessageAction.Edit(messageContent.newContent, message.id)
         )
+
+    override fun fromEntitiesToLocalNotifications(
+        list: List<NotificationMessageEntity>,
+        messageSizePerConversation: Int,
+        mapMessage: (NotificationMessageEntity) -> LocalNotificationMessage?
+    ): List<LocalNotification.Conversation> =
+        list.groupBy { it.conversationId }
+            .map { (conversationId, messages) ->
+                LocalNotification.Conversation(
+                    // todo: needs some clean up!
```

Comment:
question: is that todo valid anymore?

Reply:
I didn't add this `TODO` and just moved the whole mapping fun into mapper, so maybe it's still not clear ;P